source/core/src/main/com/csse3200/game/components/quests/QuestManager.java

```diff
@@ -51,41 +58,133 @@ public QuestManager(Entity player) {
     /**
      * Sets up the tasks for the quests and dialogues.
      */
-    private void testQuests() {
 
-        //creates test tasks
+    private Task[] createTasks() {
         Task stepsTask = new Task("steps", "Take your first steps", "Just start moving!", 1, 0, false, false);
         Task attackTask = new Task("attack", "Swing your first sword", "Just Attack!", 1, 0, false, false);
         Task testKangaTask = new Task("spawnKangaBoss", "He is Coming...", "RUN", 1, 0, false, false);
+        Task talkToGuide = new Task("talkToGuide", "Talk to the cow", "Speak with the Guide to start your journey.", 1, 0, false, false);
+        Task followCowsTeachings = new Task("followCowsTeachings", "Complete further quests", "Complete first steps and 2 step quest or a combat quest", 1, 0, false, false);
+        Task collectPotions = new Task("item collection task successful", "Collect Potions", "Collect 5 defense potions scattered around the kingdom.", 1, 0, false, false);
+        Task listenAdvice = new Task("listenToGuide", "Visit cow again", "Go visit the cow!", 1, 0, false, false);
+        Task exploreWild = new Task("exploration", "Explore and ask around", "Ask other animals about Kanga!", 1, 1, false, false);
+        Task retrieveWeapon = new Task("retrieveWeapon", "Complete the minigame", "Play the snake minigame!", 1, 0, false, false);
+
+        return new Task[] {
+                stepsTask, attackTask, testKangaTask, talkToGuide, followCowsTeachings,
+                collectPotions, listenAdvice, exploreWild, retrieveWeapon
+        };
+    }
 
-        //creates single task quest
-        List<Task> tasks = List.of(stepsTask);
-        QuestBasic firstStepsQuest = new QuestBasic("First Steps","Take your first steps in this world!", tasks, false,null,null, false, false, 0);
+    /**Sets up the dialogue for quests. */
+    private Map<DialogueKey, String[]> createQuestDialogues() {
+
+        String[] cowInitialDialogue = {
+                "Moo there, adventurer! Welcome to the kingdom.",
+                "We’ll be your guides but before you can roam free you must complete the first steps and 2 step quests."
```

Comment:
Why is there dialogue in multiple places (here and json)?

Reply:
Kinda hard coded it since we couldn't integrate with NPC team this sprint.